File: typescript-javascript/algorithms/easy/staircase.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-05-19
 * Last modified: 2026-06-11
 * @description Staircase - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/staircase/problem}
 */
//---------- functions ----------
/**
 * Prints a staircase of given size.
 * @param {number} n - The size of the staircase.
 */
function staircase(n: number): void
{
    for(let i = 1; i <= n-1; i++)
    { console.log(" ".repeat(n-i) + "#".repeat(i)); }
    
    console.log("#".repeat(n));
}//end staircase()

File: python/algorithms/easy/staircase.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Staircase - HackerRank
Test description: https://www.hackerrank.com/challenges/staircase/problem
"""
#---------- functions ----------
def staircase(n: int) -> None:
    """
    Given a positive integer, print a staircase of size n.

    Parameters:
        n (int): The size of the staircase.
    """
    for i in range(1, n + 1):
        print(" " * (n - i) + "#" * i)